### DISCUSSION TITLE
Hierarchical menu sub level indicator

### DISCUSSION THREAD

**Discussion by morakniv1891** [algolia/instantsearch#5098] [Ideas]
We need a way to be able to tell if categories have subcategories (for ui/ux/styling) before selecting that category in the hierarchical menu. A simple bool on the item would do so we can add a chevron or similar if the category has subcategories

**Comment by Haroenv** [comment]
Unfortunately this would require a query per entry in the hierarchical menu, as we don't know which levels are available until they are selected. If you are interested in knowing more information, you can make a custom widget that does the network requests differently

**Reply by morakniv1891** [reply]
Ok, I might be missing something here but it seems to me like you actually do know the levels. If I look at the results from the search, everything that's needed is right there?
![Annotation 2022-09-06 105929](https://user-images.githubusercontent.com/56967749/188593706-6eb5e45f-9181-45c3-8a38-2ea964a33a2b.jpg)

**Reply by Haroenv** [reply]
Unfortunately without applying an actual filter for each item (like Knivar and Tillbehör) in a separate request there's no way to know in advance exhaustively all the filters. In your case there's very few items, but in real life you probably have more

**Reply by morakniv1891** [reply]
Is there any way I can access the latest results object within a custom hierarchicalMenu widget made with connectHierarchicalMenu then? Because it looks like this will solve my problem.

**Reply by morakniv1891** [reply]
To clarify, the latest version that's being used for displaying results. I guess there is a cached version that gets reused if the search go back to some previous state?

**Reply by Haroenv** [reply]
inside connnectHierarchicalMenu you can use `instantSearchInstance.mainIndex.getResults()`

**Reply by morakniv1891** [reply]
Thank you very much! Seems to work like a charm. :)
